force-app/main/default/classes/LicensingUtil.cls

```diff
@@ -0,0 +1,65 @@
+public with sharing class LicensingUtil {
+	
+ 	private static String currentNamespace = getNamespace();
+    private static Set<String> packageNamespaces = new Set<String> {'plpdp_mynamespace1', 'IsvlTdxDemo'};
+    
+	
+    public static boolean hasAccessToLCP(String customPerm) {
+        // TODO: Confirm that the customPerm is not null, exists, and is a Licensed Custom Perm - otherwise throw exception
+        
+        /*
+        System.debug('debug Test.isRunningTest ' + Test.isRunningTest());
+        System.debug('debug isDevOrg ' + isDevOrg());
+        System.debug('debug isPackagingOrg ' + isPackagingOrg());
+        System.debug('debug userHasPackageLicense ' + userHasPackageLicense());
+        System.debug('debug FeatureManagement.checkPermission ' + FeatureManagement.checkPermission(customPerm));
+        System.debug('debug getNamespace ' + getNamespace());
+        System.debug('debug getPackageId ' + getPackageId());
+        */
+        
+        if(Test.isRunningTest() ||                           // bypass access check for automated tests
```

Comment:
those steps should not work, as unconstrained perm sets are validated upon assignment to confirm the user has entitlement (via a license) for whatever settings the unconstrained perm set contains. at step 4 there should be an error saying the user doesn't have the license for that LCP. if it didn't work this way, then Subscribers would be able to bypass ISV's licensing 

ISVs do sometimes do manual testing in dev org during implementation, but best practice is to only do real QA with a package installed org. But in our case that kind of dev-side manual testing is not possible, since the access check does require the user (in dev org) to have LCP assigned. But it's impossible to assign an LCP to a user in dev org, since no license exists for LCP entitlement. All perm set grants with LCP would fail